File: otel/GenevaLoggingLimit/GenevaLoggingLimit-DotNet/Tests/GenevaTest.cs
using System.Diagnostics;

namespace GenevaLoggingLimit_DotNet.Tests;

/// <summary>
/// Mirrors the C++ GenevaTest.cpp — entry points that wire up
/// the Geneva/ETW pipeline and run the shared test functions.
/// </summary>
internal static class GenevaTest
{
    private const string ServiceName = "Geneva-Otlp-Test-App";
    private const string TracerName = "MSATraceV1";
    // "DatapointLogLib" is the logger category — the Geneva exporter maps it
    // to the "MSAWarmPathV2" table via TableNameMappings.
    private const string LoggerCategory = "DatapointLogLib";

    public static void TestLogWithGeneva()
    {
        using var pipeline = OTelPipelineFactory.CreateGeneva(ServiceName);
        pipeline.Start();

        using var activitySource = new ActivitySource(TracerName);
        var logger = pipeline.LoggerFactory.CreateLogger(LoggerCategory);

        // TestRunner.TestLog(logger, activitySource);
        TestRunner.TestLogAttributeLimit(logger, activitySource);

        pipeline.Shutdown();
    }

    public static void TestTraceWithGeneva()
    {
        using var pipeline = OTelPipelineFactory.CreateGeneva("OTel-Pipe-With-ETW");
        pipeline.Start();

        using var activitySource = new ActivitySource(TracerName);
        var logger = pipeline.LoggerFactory.CreateLogger(LoggerCategory);

        TestRunner.TestTrace(logger, activitySource);

        pipeline.Shutdown();
    }
}

File: otel/GenevaLoggingLimit/GenevaLoggingLimit-Cpp/test/GenevaTest.cpp
#include <string>

#include "../IOtelPipeline.h"
#include "GenevaTest.h"
#include "test.h"

namespace {
  const std::string serviceName = "Geneva-Otlp-Test-App";
  //const std::string tracerName = "Geneva-Test-Library";
  //const std::string loggerName = tracerName;
  const std::string tracerName = "MSATraceV1";
  const std::string loggerName = "MSADataPointLogger";
}


namespace MsaLab { namespace Details
{

  void TestLogWithGeneva()
  {
    auto otel = MsaLab::Api::CreateOTelPipeline(serviceName);
    otel->Start();

    auto tracer = MsaLab::Api::GetTracer(tracerName);
    auto logger = MsaLab::Api::GetLogger(loggerName, "DatapointLogLib");

    // TestLog(logger, tracer);

    TestLogAttributeLimit(logger, tracer);

    otel->Shutdown();
  }

void TestTraceWithGeneva()
{
  auto otelPipeline = MsaLab::Api::CreateOTelPipeline("OTel-Pipe-With-ETW");
  otelPipeline->Start();

  auto tracer = MsaLab::Api::GetTracer(tracerName);
  auto logger = MsaLab::Api::GetLogger(loggerName, "DatapointLogLib");

  TestTrace(logger, tracer);
  otelPipeline->Shutdown();
}

} // namespace Details
} // namespace MsaLab

